includes/activity/class-base-object.php

```diff
@@ -55,7 +55,7 @@
  * @method string[]|null        get_summary_map()        Gets the summary map property of the object.
  * @method array[]|null         get_tag()                Gets the tag property of the object.
  * @method string[]|string|null get_to()                 Gets the primary recipients of the object.
- * @method string               get_type()               Gets the type of the object.
+ * @method string[]|string      get_type()               Gets the type of the object.
```

Comment:
Is there an example you could share for a type that's an array?

Reply:
I can't recall. But I heard someone having received such data in federation, where he would just pick the first item then. All I know is that the spec allows it. Though it's really common not to use it.

Reply:
Hm, interesting. This might need more work than just the docs update to make it work.

I'm on the fence about leaving it until we run into a bug with this, vs documenting and not handling it.

Reply:
Totally true. But it could be a nice reminder :)

During one of the recent conferences we discussed that sometimes multi-typing might be handy, but everyone opted not to do so, because the fear is too strong too break federation with others that might not be able to handle it, or have appropriate fallbacks!

Reply:
> But it could be a nice reminder :)

My concern is more about expectations and confidence. If we document `type` as possibly being an array we should make sure all corresponding code accounts for it to avoid ambiguity in the future and avoid creating areas  of code that people don't want to touch because it could break in unexpected ways.

Reply:
```suggestion
 * @method string               get_type()               Gets the type of the object.
```
I'm no longer on the fence 🙂 Let's keep it as-is for now and account for any activities that use it when we come across them